test/unit/radio.js

```diff
@@ -86,4 +90,44 @@ describe('MaterialRadio', function () {
     }())).to.equal('mdl-radio mdl-js-radio is-upgraded is-checked');
   });
 
+  it('should update related radios on one changing', function() {
+    var radios = [];
+    radios.push(createRadioWithValues('test', 'one'));
+    radios.push(createRadioWithValues('test', 'two'));
+    radios.push(createRadioWithValues('test', 'three'));
+    radios.push(createRadioWithValues('tester', 'A'));
+    radios.push(createRadioWithValues('tester', 'B'));
+    radios.push(createRadioWithValues('tester', 'C'));
+    var container = document.createElement('div');
+    radios.forEach(function(item) {
+      container.appendChild(item);
+      componentHandler.upgradeElement(item);
+    });
+    document.body.appendChild(container);
+
+    // Prepare the change event for manual firing.
+    // Used to trigger sibling checking as if UX triggered.
+    var changeEvent = document.createEvent("HTMLEvents");
+    changeEvent.initEvent("change", false, true);
+
+    // Check that all inputs are in a clean state
+    Array.prototype.splice.call(document.querySelectorAll('[type="radio"]')).forEach(function(item) {
+      expect(item.className).to.equal('mdl-radio mdl-js-radio is-upgraded');
+    });
+
+    radios[0].MaterialRadio.check();
+    radios[0].querySelector('input').dispatchEvent(changeEvent);
```

Comment:
It shouldn't break existing code at all. But JS itself doesn't trigger the change event when modifying checked state, you need to manually fire it.

I think for now we should minimize the modifications made here and keep it as-is. Open a new issue for researching adding event fires in form methods where appropriate.